driverapi/driverapi.go

```diff
@@ -72,6 +72,11 @@ type Driver interface {
 	// only invoked for the global scope driver.
 	EventNotify(event EventType, nid string, tableName string, key string, value []byte)
 
+	// DecodeTableEntry passes the driver a key, value pair from table it registered
+	// with libnetwork and expects the driver to return a {endpoint ID, info string}
```

Comment:
`Actually, could there be a case where the first element of the returned tuple would differ from the input parameter key ?` - Key is what the driver passed to networkdb. It could be anything the driver uses to identify the endpoint. So the driver should be able to get the endpoint ID and return it along with a human readable info string. In the case of overlay driver key happens to be the endpoint ID. 

With the current APIs driver doesn't convey what a given table is used for. Even though the `DecodeTableEntry` API is generic without knowing what item we are dealing with it can't be presented meaningfully in the UX. Since the current usage is only for endpoint (driver creates only one entry during the join) we are using it to associate with the libnetwork's endpoint info. If we allow the driver to create arbitrary tables we have to make some changes to the APIs. So the comment is valid and correct for the current usage.